### PR TITLE
DOC Fixing the issue with return_X_y in documentation for fetch datasets (ACS Income dataset)

### PR DESCRIPTION
**Pull request by Tizpaz**
## Description

This is my good first issue:) The goal is to address Issue #1205 created by @romanlutz. This is just example for one dataset (ACS Income dataset). Once confirmed, will do for the other datasets.

## Tests

- [X] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [] user guide added or updated
- [X ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Review by romanlutz — COMMENTED** [review]
I filed this issue because I wasn't quite sure what it should be like, but certainly not the way it was before 🤣 So with that in mind, I'm glad you started the work on it! What you have is almost correct, but there is something missing in case `as_frame=True`.

Perhaps we should follow the way `scikit-learn` documents this since we're just using their `fetch_openml` function: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_openml.html#sklearn.datasets.fetch_openml

image

[ @Tizpaz sorry I should have mentioned this in the issue description. A good example of an underspecified issue!]

**Comment by Tizpaz** [comment]
Thanks @romanlutz for following-up. Sure, let me update my pull request following the same return structure as scikit-learn fetch_openml. Please feel free to keep the pull request open until we are more sure about the changes.

File: fairlearn/datasets/_fetch_acs_income.py
Comment (romanlutz) [inline fairlearn/datasets/_fetch_acs_income.py:99]:
```suggestion
    We follow the convention of :func:`sklearn.datasets.fetch_openml` to document this function.
```
I believe this little change should generate link to the `fetch_openml` API docs on the scikit-learn page. If not, we can try adding `~` before `sklearn` just like with `Bunch` earlier in this file.

File: fairlearn/datasets/_fetch_acs_income.py
Comment (romanlutz) [inline fairlearn/datasets/_fetch_acs_income.py:95]: Thanks for making this just like the sklearn fetch_openml API docs!

I'm mildly surprised that they don't mention the data type for data and target (ndarray vs DataFrame/Series). I guess they do mention it above with data and target so it's fine by me like this.

@fairlearn/fairlearn-maintainers any thoughts? If you're fine with this please upvote so that @Tizpaz can move on to the other `fetch_*` functions in the datasets module.

Comment (romanlutz) [reply fairlearn/datasets/_fetch_acs_income.py:95]: @Tizpaz if you prefer to have this PR for ACS Income only that's fine, too! It's entirely up to you...

Comment (Tizpaz) [reply fairlearn/datasets/_fetch_acs_income.py:95]: Thanks, @romanlutz. I am fine with doing this to all datasets in `fetch_*`, especially for consistency (we should also be fair to all of our datasets!!).

File: fairlearn/datasets/_fetch_acs_income.py
Comment (MiroDudik) [inline fairlearn/datasets/_fetch_acs_income.py:97]: I don't think that we need this note. But, if you want to acknowledge the relationship with fetch_openml, we could say that our API largely follows the API of `sklearn.datasts.fetch_openml`.

Comment (Tizpaz) [reply fairlearn/datasets/_fetch_acs_income.py:97]: Sure. I can revise the comments in the note. I will send the updated pull. Let me know if that sounds ok to move forward:)

File: fairlearn/datasets/_fetch_acs_income.py
Comment (adrinjalali) [reply fairlearn/datasets/_fetch_acs_income.py:95]: yeah data and target are explained above, I'm happy as is.

**Comment by Tizpaz** [comment]
Alright! I just updated all the fetch functions in the datasets. Please feel free to merge this pull request if everything looks good.

File: fairlearn/datasets/_fetch_boston.py
Comment (adrinjalali) [inline fairlearn/datasets/_fetch_boston.py:120]:
```suggestion
    -----
```
the separator should have the same number of chars as the title above it.

**Review by adrinjalali — APPROVED** [review]
Otherwise LGTM>

**Comment by hildeweerts** [comment]
@Tizpaz currently [flake8 fails](https://github.com/fairlearn/fairlearn/actions/runs/4355859090/jobs/7650206447) due to several trailing whitespaces and blank lines containing whitespaces. Could you resolve those? It would be great to get this PR merged! :D

**Comment by Tizpaz** [comment]
Hi @hildeweerts, sure, I can do so. Let me sure that I have the token from @adrinjalali to move forward.

**Comment by adrinjalali** [comment]
I've fixed the linting issues now, this should be ready to be merged

**Comment by adrinjalali** [comment]
I've stopped pushing to your branch now @Tizpaz

**Comment by Tizpaz** [comment]
Thank you, @adrinjalali. I will check-in later (after others' comments) to see if more things need to be done.

**Pull request merged**
